sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -284,6 +285,12 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
    * @since 1.4.0
    */
   def save(path: String): Unit = {
+    if (!df.sparkSession.sessionState.conf.legacyPathOptionBehavior &&
+        extraOptions.contains("path") && path.nonEmpty) {
```

Comment:
The `path` here is a String, do we really need to check `path.nonEmpty`?

Reply:
Good catch. I think we need to remove the check. The output is more confusing since `path` option is also set:
```scala
scala> Seq(1).toDF.write.option("path", "/tmp/path1").parquet("")
java.lang.IllegalArgumentException: Can not create a Path from an empty string
  at org.apache.hadoop.fs.Path.checkPathArg(Path.java:168)
```

I will create a PR for this.

Reply:
Created https://github.com/apache/spark/pull/29697